clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -343,15 +343,34 @@ static void collectSYCLAttributes(Sema &S, FunctionDecl *FD,
   if (!FD->hasAttrs())
     return;
 
-  llvm::copy_if(FD->getAttrs(), std::back_inserter(Attrs), [](Attr *A) {
-    // FIXME: Make this list self-adapt as new SYCL attributes are added.
-    return isa<IntelReqdSubGroupSizeAttr, IntelNamedSubGroupSizeAttr,
-               ReqdWorkGroupSizeAttr, SYCLIntelKernelArgsRestrictAttr,
-               SYCLIntelNumSimdWorkItemsAttr,
-               SYCLIntelSchedulerTargetFmaxMhzAttr,
-               SYCLIntelMaxWorkGroupSizeAttr, SYCLIntelMaxGlobalWorkDimAttr,
-               SYCLIntelNoGlobalWorkOffsetAttr, SYCLSimdAttr>(A);
-  });
+  // Attributes that should be propagated from device functions to a kernel
+  // in SYCL 1.2.1.
+  if (S.getASTContext().getLangOpts().getSYCLVersion() <
```

Comment:
Note: this will return `SYCL_None` unless `-syclstd=` is specified on the cc1 command line, so the behavior here may be incorrect in that case (I don't think we want to treat that as using the 1.2.1 spec).

Reply:
@AaronBallman, should we pass -sycl-std=2017 instead?

-sycl-std=2017: Propagate attribute
-sycl-std=2020: No Propagation

Reply:
I think we have to decide what it means when passing `-cc1 -fsycl-is-device|host` but not passing `-sycl-std=`. On the driver side of things, calling `clang -fsycl whatever.cpp` will produce a cc1 command line that includes `-sycl-std=2020`. On the FE side of things, I think we should default to 2020 as well, but perhaps @mdtoguchi has opinions? We could either set the SYCL version to SYCL_2020 manually, or I suppose we could add a `SYCL_Default` enumerator to signify "nothing was passed, but you're still in SYCL mode", etc.

Reply:
I did the implementation work on this in: https://github.com/intel/llvm/pull/3702 and once that lands, this code is fine.

Reply:
> I did the implementation work on this in: #3702 and once that lands, this code is fine.

Thanks @AaronBallman.